### PR TITLE
BAEL-5419: List vs ArrayList in Java + tests

### PR DESCRIPTION
**Pull request by palani-a**
Committed for BAEL-5419: List vs ArrayList in Java + tests

Draft Link: https://drafts.baeldung.com/wp-admin/post.php?post=129943&action=edit

### CODE REVIEW

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ListVsArrayListUnitTest {
+
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+    private ArrayList<String> passengerList = new ArrayList<>();
+
+    @BeforeEach
+    public void setUp() throws Exception {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java:21]: Need not be `public`

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        assertNotNull(actual);
+        assertThat(actual.get(0)).isEqualTo(passengerList.get(0));
+    }
+
+    @Test
+    void givenLocale_whenUsingStreams_thenReturnsListType() {
+        List<String> actual = ListVsArrayList.getCountries();
+
+        assertNotNull(actual);
+        assertThat(actual.size()).isEqualTo(Locale.getISOCountries().length);
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java:69]: Should use `assertThat(actual).hasSize(...)`

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        assertNotNull(actual);
+        assertThat(actual).isEqualTo(passengerList.toString());
+    }
+
+    @Test
+    void givenArrayList_whenCalledWithCollectionsFunction_thenReturnsListType() {
+        List<String> actual = ListVsArrayList.getNames(passengerList);
+
+        assertNotNull(actual);
+        assertThat(actual.toString()).isEqualTo(passengerList.toString());
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java:53]: Use `assertThat(...).hasToString(...)`

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        assertThat(actual).isEqualTo(passengerList.toString());
+    }
+
+    @Test
+    void givenArrayList_whenCalledUsingListTypeParam_thenPrintsValues() {
+        List<String> list = new ArrayList<>(passengerList);
+        ListVsArrayList.printNamesListType(list);
+        String actual = outputStreamCaptor.toString().trim();
+
+        assertNotNull(actual);
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java:44]: A wasted assertion

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.jupiter.api.Test;
+
+class ListVsArrayListUnitTest {
+
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+    private ArrayList<String> passengerList = new ArrayList<>();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        System.setOut(new PrintStream(outputStreamCaptor));
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java:22]: Unit testing things that do System.out is pretty ugly. This test also fails to restore system.out afterwards.

You could use SystemStubs to handle this in Junit 5 - https://github.com/webcompere/system-stubs#using-systemerr-systemout-and-systemerrandout - where you can then use the `SystemOut` as a stub a bit like this - https://github.com/webcompere/system-stubs/blob/main/system-stubs-jupiter/src/test/java/uk/org/webcompere/systemstubs/jupiter/SystemStubsExtensionTest.java#L22 

But overall, why make code for an example that need tests this tricky to construct?

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java (hunk 1 above)
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java]: Fixed the test

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java (hunk 2 above)
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java]: Fixed the tests

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java (hunk 3 above)
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java]: Fixed the tests

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java (hunk 4 above)
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java]: Fixed the tests

File: core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java (hunk 5 above)
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java]: Removed System.out statements and used return statements to test better.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-collections-list-3/src/main/java/com/baeldung/list/listvsarraylist/Passenger.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+public class Passenger {
+
+    private String name;
+    private int age;
+    private String source;
+    private String destination;
+
+    public Passenger(String name, int age, String source, String destination) {
+        super();
```
[the remark this answers is not in the document]
Comment (palani-a) [reply core-java-modules/core-java-collections-list-3/src/main/java/com/baeldung/list/listvsarraylist/Passenger.java]: Removed super() call.

**Pull request merged**